api/v1alpha08/kogitoserverlessplatform_types.go

```diff
@@ -0,0 +1,157 @@
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ConfigurationSpec represents a generic configuration specification
+type ConfigurationSpec struct {
+	// Type represents the type of configuration, ie: property, configmap, secret, ...
+	Type string `json:"type"`
+	// Value the value to assign to the configuration (syntax may vary depending on the `Type`)
+	Value string `json:"value"`
+}
+
+const (
+	// KogitoServerlessPlatformKind is the Kind name of the KogitoServerlessPlatform CR
+	KogitoServerlessPlatformKind string = "KogitoServerlessPlatform"
+)
+
+// PlatformCluster is the kind of orchestration cluster the platform is installed into
+type PlatformCluster string
+
+const (
+	// PlatformClusterOpenShift is used when targeting an OpenShift cluster
+	PlatformClusterOpenShift PlatformCluster = "OpenShift"
+	// PlatformClusterKubernetes is used when targeting a Kubernetes cluster
+	PlatformClusterKubernetes PlatformCluster = "Kubernetes"
+)
+
+// PlatformBuildStrategy specifies how the Build should be executed.
+// It will trigger a Kaniko pod that will take care of producing the expected image containing the Quarkus/Kogito runtime.
+// +kubebuilder:validation:Enum=pod
+type PlatformBuildStrategy string
+
+const (
+	// BuildStrategyPod performs the build in a `Pod` (will schedule a new builder ephemeral `Pod` which will take care of the build action).
+	BuildStrategyPod PlatformBuildStrategy = "pod"
+)
+
+// PlatformBuildStrategies is a list of strategies allowed for the build
+var PlatformBuildStrategies = []PlatformBuildStrategy{
+	BuildStrategyPod,
+}
+
+// BuildPublishStrategies the list of all available publish strategies
+var BuildPublishStrategies = []api.PlatformBuildPublishStrategy{
+	api.PlatformBuildPublishStrategyKaniko,
+}
+
+// KogitoServerlessPlatformSpec defines the desired state of KogitoServerlessPlatform
+type KogitoServerlessPlatformSpec struct {
+	// Cluster what kind of cluster you're running (ie, plain Kubernetes or OpenShift)
+	Cluster PlatformCluster `json:"cluster,omitempty"`
+	// Build specify how to build the Workflow
+	Build api.BuildSpec `json:"build,omitempty"`
+	//BuildPlatform specify how it is the platform where we want to build the Workflow
+	BuildPlatform api.PlatformBuildSpec `json:"platform,omitempty"`
+	// Configuration list of configuration properties to be attached to all the Workflow built from this Platform
+	Configuration ConfigurationSpec `json:"configuration,omitempty"`
+}
+
+// PlatformPhase is the phase of an Platform
+type PlatformPhase string
+
+const (
+	// PlatformPhaseNone when the KogitoServerlessPlatform does not exist
+	PlatformPhaseNone PlatformPhase = ""
+	// PlatformPhaseCreating when the KogitoServerlessPlatform is under creation process
+	PlatformPhaseCreating PlatformPhase = "Creating"
+	// PlatformPhaseWarming when the KogitoServerlessPlatform is warming (ie, creating Kaniko cache)
+	PlatformPhaseWarming PlatformPhase = "Warming"
```

Comment:
Preparing ?

Reply:
Instead of "Creating" ?

Reply:
instead of Warming. I would say this is Kaniko terms. But we should be more global.
You create and then you prepare ? 
Anyway, as we are in Alpha state, we can keep the Warming for now and change it later if we see something that fit more